geomesa-accumulo/geomesa-accumulo-datastore/src/main/scala/org/locationtech/geomesa/accumulo/util/GeoMesaBatchWriterConfig.scala

```diff
@@ -24,6 +24,42 @@ object GeoMesaBatchWriterConfig extends Logging {
       num <- Try(java.lang.Long.parseLong(p)).toOption
     } yield num
 
+  protected [util] def fetchMemoryProperty(prop: String): Option[Long] =
+    for {
+      p <- Option(System.getProperty(prop))
+      num <- parseMemoryProperty(p)
+    } yield num
+
+  protected [util] def parseMemoryProperty(prop: String): Option[Long] = {
+
+    //Scala regex matches the whole string, the leading ^ and trailing $ is implied.
+    val matchSuffix = """(\d+)([A-Za-z]+)?""".r
+
+    //Define suffixes based on powers of 2
+    val suffixMap = Map(
+      "K" -> 1024l,
+      "k" -> 1024l,
+      "M" -> 1024l * 1024l,
+      "m" -> 1024l * 1024l,
+      "G" -> 1024l * 1024l * 1024l,
+      "g" -> 1024l * 1024l * 1024l
+    )
+
+    prop match {
+      case matchSuffix(number, null) =>
+        for {
+          num <- Try(java.lang.Long.parseLong(number)).toOption
+        } yield num
+      case matchSuffix(number, suffix) =>
+        if(suffixMap.contains(suffix))
+          for {
+            num <- Try(java.lang.Long.valueOf(number.toLong * suffixMap(suffix))).toOption
+          } yield num
```

Comment:
For comprehensions are good, but I'm pretty sure this can be simplified to just 
```
Try(java.lang.Long.valueOf(number.toLong * suffixMap(suffix))).toOption
```
Likewise, the above one can be simplified.